src/nvim/CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ if(WIN32)
   # tell MinGW compiler to enable wmain
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -municode")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -iframework CoreFoundation")
```

Comment:
Should this be ` -framework` (without `i`) ?

Reply:
In that case I'd get this message for each C file:

```
[278/282] Building C object src/nvim/CMakeFiles/nvim.dir/viml/parser/expressions.c.o
clang: warning: -framework CoreFoundation: 'linker' input unused [-Wunused-command-line-argument]
```

I think we need @jszakmeister to tell us about the root intention. :>

Reply:
I think we need to drop that line.  I forgot to remove it after I found the real culprit (the missing `-isysroot`).  The hope was that it would help with pulling in CoreFoundation for the generation step, but as you see here, it didn't.

This was definitely a weird problem overall.  It turns out there's a difference between executing `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang` and `/usr/bin/clang`.  CMake was picking up the former and with that it left off a key include path.  Why?  I dunno. :-(

Either way this line can be dropped as it didn't fix the problem.  Sorry about the troubles... it was definitely a weird bug to track down.